powertools-kafka/src/main/java/software/amazon/lambda/powertools/kafka/serializers/KafkaProtobufDeserializer.java

```diff
@@ -56,44 +56,53 @@ protected <T> T deserializeObject(byte[] data, Class<T> type) throws IOException
         }
     }
 
-    private Message deserializeWithMessageIndexHandling(byte[] data, Parser<Message> parser) throws IOException {
+    private <T> T defaultDeserializer(byte[] data, Class<T> type) throws IOException {
         try {
-            LOGGER.debug("Attempting to deserialize as standard protobuf data");
-            return parser.parseFrom(data);
+            LOGGER.debug("Using default Protobuf deserializer");
+            Parser<Message> parser = (Parser<Message>) type.getMethod("parser").invoke(null);
+            Message message = parser.parseFrom(data);
+            return type.cast(message);
         } catch (Exception e) {
-            LOGGER.debug("Standard protobuf parsing failed, attempting Confluent message-index handling");
-            return deserializeWithMessageIndex(data, parser);
+            throw new IOException("Failed to deserialize Protobuf data.", e);
         }
     }
 
-    private Message deserializeWithMessageIndex(byte[] data, Parser<Message> parser) throws IOException {
-        CodedInputStream codedInputStream = CodedInputStream.newInstance(data);
-
+    private <T> T confluentDeserializer(byte[] data, Class<T> type) throws IOException {
         try {
-            // https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/index.html#wire-format
-            // Read the first varint - this could be:
-            // 1. A single 0 (simple case - first message type)
-            // 2. The length of the message index array (complex case)
-            int firstValue = codedInputStream.readUInt32();
 
-            if (firstValue == 0) {
-                // Simple case: Single 0 byte means first message type
-                LOGGER.debug("Found simple message-index case (single 0), parsing remaining data as protobuf");
-                return parser.parseFrom(codedInputStream);
-            } else {
-                // Complex case: firstValue is the length of the message index array
-                LOGGER.debug("Found complex message-index case with array length: {}, skipping {} message index values",
-                        firstValue, firstValue);
-                for (int i = 0; i < firstValue; i++) {
-                    codedInputStream.readUInt32(); // Skip each message index value
+            LOGGER.debug("Using Confluent Deserializer");
+            Parser<Message> parser = (Parser<Message>) type.getMethod("parser").invoke(null);
+            ByteBuffer buffer = ByteBuffer.wrap(data);
+            int size = ByteUtils.readVarint(buffer);
+
+            // Only if the size is greater than zero, continue reading varInt. based on Glue Proto deserializer implementation
```

Comment:
Comments needs to be updated to "Confluent".

Reply:
https://github.com/aws-powertools/powertools-lambda-java/pull/1907/commits/f4f66dd0507dcde26c59d89cbbd31f346d559ac7